### PR TITLE
Remove Clm51 references throughout or change to Clm60

### LINKED ISSUE #2561: NOANTHRO qualifier NOT working in ctsm5.2.005 for CLM51 physics (only for CLM50 or CLM60)
### Brief summary of bug

This is from Jiang in the Paleo group:

> Please note that run 091 is using CLM51%BGC-CROP in its compset. Based on that, my LGM simulation is using CLM51%SP-NOANTHRO.

### General bug information

**CTSM version you are using:** ctsm5.2.005 (in cesm2_3_beta17)

**Does this bug cause significantly incorrect results in the model's science?** Yes

**Configurations affected:** CLM51%SP-NOANTHRO
### Important output or errors that show the problem

> See here
> /glade/campaign/cesm/cesmdata/cseg/runs/cesm2_0/b.e23_alpha17f.BLT1850.ne30_t232.091
> /glade/u/home/jiangzhu/cesm/cases/b.e23.BLT1850.ne30pg3_t232.cesm2_3_b17.21ka.01

### PR DESCRIPTION
**Pull request by slevis-lmwg**
### Description of changes

TODO list appears in the corresponding issue #2379.

### Specific notes

Contributors other than yourself, if any:
@ekluzek 

CTSM Issues Fixed (include github issue #):
Most of what's need in  #2379
Resolves #2561
 CMEPS, CISM and mizuRoute still need to be updated for #2379 to be finalized

Are answers expected to change (and if so in what way)?
No.

Any User Interface Changes (namelist or namelist defaults changes)?
Removing Clm51 references throughout.

Does this create a need to change or add documentation? Did you do so?
Yes.

Testing performed, if any:
On derecho:
- PASS make black and make lint
- PASS ./run_ctsm_py_tests -s and ./run_ctsm_py_tests -u
- PASS ./build-namelist_test.pl
- OK aux_clm without mods to the submodules, yet

### CODE REVIEW

**Comment by slevis-lmwg** [comment]
aux_clm OK on derecho before rebasing from master to the cesm3_0_beta04_changes branch. Not sure, yet, whether I should rerun again after rebasing.

aux_clm IN PROG on izumi after rebasing to the cesm3_0_beta04_changes branch.

File: bld/unit_testers/build-namelist_test.pl [hunk 1]
```diff
@@ -600,10 +600,6 @@ sub cat_and_create_namelistinfile {
                                      namelst=>"use_crop=.true.",
                                      phys=>"clm4_5",
                                    },
-     "LeungDust_WO_Prigent"      =>{ options=>" -envxml_dir . -bgc sp",
-                                     namelst=>"use_prigent_roughness=.true.",
-                                     phys=>"clm5_1",
```
Comment (ekluzek) [inline bld/unit_testers/build-namelist_test.pl:605]: Just update this to clm6_0

File: bld/unit_testers/build-namelist_test.pl [hunk 2]
```diff
@@ -966,16 +962,6 @@ sub cat_and_create_namelistinfile {
...
-     "sasuspinupWOsoilmatx"      =>{ options=>"-envxml_dir . -bgc bgc -clm_accelerated_spinup sasu",
-                                     namelst=>"use_soil_matrixcn=.false.,use_matrixcn=.false.",
-                                     GLC_TWO_WAY_COUPLING=>"TRUE",
-                                     phys=>"clm5_1",
-                                   },
-     "sasuspinupWOCN"            =>{ options=>"-envxml_dir . -bgc sp  -clm_accelerated_spinup sasu",
-                                     namelst=>"",
-                                     GLC_TWO_WAY_COUPLING=>"TRUE",
-                                     phys=>"clm5_1",
-                                   },
```
Comment (ekluzek) [inline bld/unit_testers/build-namelist_test.pl:978]: Update these to clm6_0 and keep.

File: cime_config/testdefs/testlist_clm.xml [hunk 3]
```diff
@@ -120,14 +120,13 @@
       <option name="comment">CESM3 development Exact restart test with change in the processor count at f10, for izumi nag</option>
     </options>
   </test>
-  <!-- this will be removed when CESM no longer needs it-->
-  <test name="ERP_D_Ld9" grid="ne30pg3_t232" compset="I1850Clm51BgcCropG" testmods="clm/clm51cam6LndTuningMode">
+  <!-- Remove when CESM no longer needs it-->
```
Comment (ekluzek) [inline cime_config/testdefs/testlist_clm.xml:123]: Just remove this comment about removing when CESM no longer needs it.

Comment (ekluzek) [reply cime_config/testdefs/testlist_clm.xml:123]: I think this one is duplicated, so remove the duplicate here.

File: cime_config/testdefs/testlist_clm.xml [hunk 4]
```diff
@@ -946,14 +945,13 @@
     </options>
   </test>
   <!-- Remove this when CAM removes the use of CLM60 in their testing and compsets -->
```
Comment (ekluzek) [inline cime_config/testdefs/testlist_clm.xml:947]: Remove this comment above.

**Review by ekluzek — APPROVED** [review]
@slevis-lmwg and I went over this and we have a couple changes to make. But, he will make those and then bring this in.

As a final thing to do do a 
git grep -i clm5_1
git grep -i Clm51
git grep -i clm5.1

and make sure everthing is handled. We did this before and we thought it was, but good to make sure again.

File: cime_config/testdefs/testlist_clm.xml (hunk 3 above)
Comment (slevis-lmwg) [reply cime_config/testdefs/testlist_clm.xml:126]: Looking at all the ne30pg3_t232 tests, I did not encounter an exact duplicate, so I'm leaving them unchanged.

**Comment by slevis-lmwg** [comment]
Testing before merging

Derecho
PASS ./build-namelist_test
PASS python testing
PEND aux_clm, ctsm_sci

Izumi
PEND aux_clm

**Comment by slevis-lmwg** [comment]
- [ ] Return the 4 Sp tests from -nofireemis to -nofire and mark EXPECTED FAIL (issue #2784).

**Pull request merged**